models/record.py

```diff
@@ -12,6 +14,40 @@ class Record(BaseModel):
 
     # optional
     country: Optional[str] = 'N/A'
+    purchase: Optional[dict] = {}
```

Comment:
@kyle-seongwoo-jun I think the code will be simpler if the purchase field is also made into a class inherited from the BaseLine mo del.